src/components/Home.tsx

```diff
@@ -0,0 +1,126 @@
+import { useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { clearToken, type AppDispatch, type RootState } from "../store.tsx";
+import { completedToDoLists } from "../constants/data.ts";
+import "bootstrap/dist/css/bootstrap.min.css";
+
+const Home = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch<AppDispatch>();
+  const token = useSelector((state: RootState) => state.auth.token);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState("");
+
+  useEffect(() => {
+    if (!token) {
+      dispatch(clearToken());
+      navigate("/");
+    }
+  }, [navigate]);
```

Comment:
Making this a side effect means, `Home` would render once and then navigate.

We could return the `Navigate` component of `react-router` to ensure navigation happens in case user is not authenticated.
```suggestion
if (!token) {
  return <Navigate to="/" />;
}
```

Also we wouldn't need to dispatch `clearToken` as we are getting the token state from redux itself.